sw/device/lib/dif/dif_uart.c

```diff
@@ -86,6 +101,8 @@ static bool uart_configure(const dif_uart_t *uart,
     return false;
   }
 
+  uart_reset(uart);
```

Comment:
I would recommend moving this after the `if` block on lines 111-113 so that the `init` function does not have any side-effects if it fails.

Reply:
Done.